src/components/webview/WebviewPanel.ts

```diff
@@ -0,0 +1,120 @@
+import { Disposable, Uri, ViewColumn, WebviewPanel, window } from 'vscode';
+import { WebviewResolver } from './WebviewResolver';
+import { WebviewMessage } from './webviewEvents';
+import { Message, MessageBus, MessageKind } from '../messageBus';
+
+type PanelOptions = Readonly<{
+	type: string;
+	title: string;
+	showOptions:
+		| ViewColumn
+		| {
+				readonly viewColumn: ViewColumn;
+				readonly preserveFocus?: boolean | undefined;
+		  };
+}>;
+
+type WebviewOptions = Readonly<{
+	extensionUri: Uri;
+	webviewName: string;
+	initialData: unknown;
+}>;
+
+export type Options = PanelOptions & WebviewOptions;
+
+export abstract class IntuitaWebviewPanel {
+	protected __panel: WebviewPanel | null = null;
```

Comment:
if it's protected, let's perhaps make it `_panel`

`x` => public
`_x` => protected
`__x` => private

does that make sense (it's loosely following the Python conventions)?

Reply:
in js its usually `_x` for private and `x` for everything else.

Reply:
we definitely need to add this to our code style guide doc